pallets/statement/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+		/// Statement idenfier marked inactive
+		RevokedStatement,
+		/// Statement idenfier not marked inactive
+		StatementNotRevoked,
+		/// Only when the author is not the controller/delegate.
+		UnauthorizedOperation,
+		/// Statement link does not exist
+		StatementLinkNotFound,
+		/// Statement Link is revoked
+		StatementLinkRevoked,
+		// Invalid creator signature
+		InvalidSignature,
+		//Statement hash is not unique
+		HashAlreadyAnchored,
+		// Expired Tx Signature
+		ExpiredSignature,
+		// Invalid Statement Identifier
+		InvalidStatementIdentifier,
+		// Invalid Schema Identifier Length
+		InvalidIdentifierLength,
+		// Statement not part of space
+		StatementSpaceMismatch,
+		//Statement digest is not unique
+		DigestHashAlreadyAnchored,
+		// Invalid transaction hash
+		InvalidTransactionHash,
+		// Metadata limit exceeded
+		MetadataLimitExceeded,
+		// Metadata already set for the entry
+		MetadataAlreadySet,
+		// Metadata not found for the entry
+		MetadataNotFound,
+		// Maximum Number of delegates reached.
+		TooManyDelegates,
+		// More than the maximum mumber of delegates.
+		TooManyDelegatesToRemove,
+		// Authorization not found
+		AuthorizationDetailsNotFound,
+		// Maximum number of activities exceeded
+		MaxStatementActivitiesExceeded,
+		// Attestation is not found
+		AttestationNotFound,
+		// Max Digests in a call
+		MaxDigestLengthExceeded,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a new statement and associates it with its
+		/// controller. The controller (issuer) is the owner of the identifier.
+		///
+		/// Arguments:
+		///
+		/// * `origin`: The origin of the call.
+		/// * `statement_digests`: Array of the digest of the statements.
+		/// * `authorization`: AuthorizationIdOf.
+		/// * `schema_id`: The schema id of the statement.
+		///
+		/// Returns:
+		///
+		/// DispatchResult
+		#[pallet::call_index(0)]
+		#[pallet::weight({0})]
+		pub fn create(
+			origin: OriginFor<T>,
+			statement_digests: Vec<StatementDigestOf<T>>,
+			authorization: AuthorizationIdOf,
+			schema_id: Option<SchemaIdOf>,
+		) -> DispatchResult {
+			let creator = <T as Config>::EnsureOrigin::ensure_origin(origin)?.subject();
+
+			let registry_id: Ss58Identifier;
+			if let Some(schema_id) = schema_id.clone() {
+				registry_id = pallet_registry::Pallet::<T>::is_a_delegate(
+					&authorization,
+					creator.clone(),
+					Some(schema_id),
+				)
+				.map_err(<pallet_registry::Error<T>>::from)?;
+			} else {
+				registry_id = pallet_registry::Pallet::<T>::is_a_delegate(
+					&authorization,
+					creator.clone(),
+					None,
+				)
+				.map_err(<pallet_registry::Error<T>>::from)?;
+			}
+
+			// now check the digests array
+			let digests_len = statement_digests.len();
+			ensure!(digests_len <= T::MaxDigestLength::get(), Error::<T>::MaxDigestLengthExceeded);
+
+			let mut event_array: Vec<StatementDigestOf<T>> = vec![];
+			for (_, statement_digest) in statement_digests.into_iter().enumerate() {
```

Comment:
This can be just `for statement_digest in statement_digests` since you are not using the index